### DISCUSSION TITLE
sys.timerloopstart

### DISCUSSION THREAD

**Discussion by jokokiswan** [openLuat/LuatOS#92] [Q&A]
how to do proper way timeloopstart?
 i set 1s, but looping timer has 20 s looping, 
if i did not call can send function before, my sys.loopstarttimer not work, am i miss something??
thank you

![Capture](https://user-images.githubusercontent.com/58895576/223020524-85849b6d-5ace-4713-b752-c24f2c186ea8.PNG)

**Comment by wendal** [comment]
show me the code

**Reply by jokokiswan** [reply]
-- Lua tools need the information of project and version
PROJECT = "uart_irq"
VERSION = "1.0.0"

log.info("main", PROJECT, VERSION)

-- Introduce necessary library files (written in lua), internal libraries do not need require
sys = require("sys")
-- require("sysplus")

local mcp2515 = require "mcp2515"
local count_can = 0

uart_id = 1
uart_len = 1024

LEDA = gpio.setup(pin.PB26, 0, gpio.PULLUP)

if wdt then
    -- Add a hard dog to prevent the program from freezing, enable this feature on supported devices
    wdt.init(1000)
    -- wdt.init(1000) -- Initialize watchdog set to 9s
    sys.timerLoopStart(wdt.feed, 1000) -- 3s feed the dog once
end
-- init_CAN()
function init_CAN()
    local mcp2515_spi = 1
    local mcp2515_cs = pin.PB14
    local mcp2515_int = pin.PB18
    spi_mcp2515 = spi.setup(mcp2515_spi, nil, 0, 0, 8, 16 * 1000 * 1000, spi.MSB,
                            1, 0)
    mcp2515.init(mcp2515_spi, mcp2515_cs, mcp2515_int, mcp2515.CAN_500Kbps)
end

function CAN_SEND() end

function uuid_() log.info(PROJECT .. ".unique_id", mcu.unique_id():toHex()) end

function get_rmc()
    log.info("GPS", libgnss.getIntLocation())
    local rmc = libgnss.getRmc()
    log.info("rmc", json.encode(rmc))
    -- log.info("rmc", rmc.lat, rmc.lng, rmc.year, rmc.month, rmc.day, rmc.hour, rmc.min, rmc.sec)
    -- rtc.set({year=rmc.year,mon=rmc.month,day=rmc.day,hour=rmc.hour,min=rmc.min,sec=rmc.sec})
end -- 两秒打印一次

-- sys.taskInit(init_uart) --init_uart()
-- uart_send()
-- sys.taskInit(init_CAN) --init_CAN()
-- uuid_()

sys.taskInit(function()
    init_CAN()
    while 1 do
        local tick2 = mcu.ticks()
        local i = 0
        for i = 1, 1, 1 do
            mcp2515.send_buffer({id = 0x7FF, ide = false, rtr = false}, 0x00,
                                0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07) -- standard frame, data frame
            -- mcp2515.send_buffer({id = 0x1FFFFFE6, ide = true, rtr = false},
            --                     0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07) -- extended frame, data frame
            -- mcp2515.send_buffer({id = 0x7FF, ide = false, rtr = true}, 0x00,
            --                     0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07) -- standard frame, remote frame
            -- mcp2515.send_buffer({id = 0x1FFFFFE6, ide = true, rtr = true}, 0x00,
            --                     0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07) -- extended frame                    
        end

        count_can = count_can + 1
        local tick1 = mcu.ticks()
        local diff_tick = tick1 - tick2
        log.info("ticks", diff_tick / 1000, "mS")
        log.info("can_tick:", count_can)
        LEDA(count_can % 2 == 0 and 1 or 0)
        sys.wait(1000)
    end
end)

-- Usercode has ended---------------------------------------------
-- always ends with this sentence
sys.run()
-- sys.run()Do not add any statement after that!!!!!

![Capture](https://user-images.githubusercontent.com/58895576/223019264-c6b99596-9f51-490a-b4c6-634b8315941b.PNG)

**Reply by jokokiswan** [reply]
if watch dog start value 9000 -> the sending loop time would be 20 s not 4 s, 
if wdt init 1000 ->
changing sys.wait(200) does not take effect. it does loop for every 4 s. 

Logging also got issue, it does only once on first run, not like before.

Board : air103
thank you
